pkg/k8s/resource/resource.go

```diff
@@ -694,13 +694,13 @@ loop:
 	}
 }
 
-func (s *subscriber[T]) getWorkItem() (e workItem, shutdown bool) {
+func (s *subscriber[T]) getWorkItem() (e WorkItem, shutdown bool) {
 	var raw any
 	raw, shutdown = s.wq.Get()
 	if shutdown {
 		return
 	}
-	return raw.(workItem), false
+	return raw.(WorkItem), false
```

Comment:
You made the `s.wg` into a typed workqueue, yet here we assign to an `any` and type assert back. Please make `raw` a `WorkItem` and remove the type assertion to make this nice.

Reply:
Good catch, fixed